lib/internal/per_context/primordials.js

```diff
@@ -21,10 +21,28 @@ const {
 // `uncurryThis` is equivalent to `func => Function.prototype.call.bind(func)`.
 // It is using `bind.bind(call)` to avoid using `Function.prototype.bind`
 // and `Function.prototype.call` after it may have been mutated by users.
-const { bind, call } = Function.prototype;
+const { apply, bind, call } = Function.prototype;
 const uncurryThis = bind.bind(call);
 primordials.uncurryThis = uncurryThis;
 
+const applyBind = bind.bind(apply);
+primordials.applyBind = applyBind;
+
+// Methods that accept a variable number of arguments, and thus it's useful to
+// also create `${prefix}${key}Apply`, which uses `Function.prototype.apply`,
+// instead of `Function.prototype.call`, and thus doesn't require iterator
+// destructuring.
+const varargsMethods = [
+  'ArrayPrototypePush',
+  'ArrayPrototypeUnshift',
+  'ArrayOf',
+  'MathMax',
+  'MathMin',
+  'MathHypot',
+  'StringPrototypeConcat',
+  'TypedArrayOf',
+];
```

Comment:
Out of curiosity, how did you came up with this list? Should we add a test to make sure it's stay updated? (if new variadic built-in are added to the spec in the future)

Reply:
I just went through the spec and added most things that accept a variable number of arguments.

---

I chose not to add `ArrayPrototypeConcat`, as it performs the spread on its own for arrays and array‑likes with a truthy `Symbol.isConcatSpreadable` property.

I also skipped adding `FunctionPrototypeCall`, as there’s already `FunctionPrototypeApply` and `ReflectApply`.

Reply:
I think it'd be worth adding a comment about the only you left out, to avoid someone opening a PR to add them later.
```js
  // 'ArrayPrototypeConcat', // performs the spread on its own for arrays and array‑likes.
```